packages/server/lib/socket-base.ts

```diff
@@ -382,9 +382,9 @@ export class SocketBase {
         })
 
         const setCrossOriginCookie = ({ cookie, url, sameSiteContext }: { cookie: SerializableAutomationCookie, url: string, sameSiteContext: SameSiteContext }) => {
-          const domain = cors.getOrigin(url)
+          const { origin } = new URL(url)
 
-          cookieJar.setCookie(automationCookieToToughCookie(cookie, domain), url, sameSiteContext)
+          cookieJar.setCookie(automationCookieToToughCookie(cookie, origin), url, sameSiteContext)
```

Comment:
does it set the `origin` or the `domain` on the cookie? I see this is just a port of the existing logic, but I wonder if it's correct. For example:

for URL `https://www.foobar.com/baz` the
* `origin`: `https://www.foobar.com`
* `domain`: `www.foobar.com`

my guess is `tough-cookie` parses this out for us but I want to double check here. Are you able to verify that?

Reply:
I'd changed the var name to more accurately portray what we're actually passing in (it's always been `origin`). The name of the parameter is `domain` on the tough cookie side of things, but the examples they use are actually the origin, not the domain: https://github.com/salesforce/tough-cookie?tab=readme-ov-file#example

Reply:
It looks like they are handling things correctly on there end and it shouldn't matter. I don't think there are existing bugs with this I just wanted to be sure that `origin` ultimately translates to `domain` on the `tough-cookie` side of things

Reply:
This is a good catch - I'll change this to `hostname` to more accurately match the [RFC that tough-cookie implements](https://www.rfc-editor.org/rfc/rfc6265.html#section-5.2.3).

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/30770/commits/fb1219cf4f0c371e7490019895c2695180eab23d